api/users/users-router.js

```diff
@@ -1,44 +1,82 @@
-const express = require('express');
+const express = require("express");
+const users = require("../users/users-model");
+const posts = require("../posts/posts-model");
+const middleware = require("../middleware/middleware");
 
 // You will need `users-model.js` and `posts-model.js` both
 // The middleware functions also need to be required
 
 const router = express.Router();
 
-router.get('/', (req, res) => {
+router.get("/", (req, res) => {
   // RETURN AN ARRAY WITH ALL THE USERS
+  users
+    .get()
+    .then((user) => {
+      res.status(200).json(user);
+    })
+    .catch((error) => {
+      res.status(400).json("Ops, something happened");
+    });
 });
 
-router.get('/:id', (req, res) => {
-  // RETURN THE USER OBJECT
+router.get("/:id", middleware.validateUserId(), (req, res) => {
```

Comment:
Marin, dar aplicatia in general merge? Eu vad ca tu invoci fiecare middleware, pe cand nu ar trebui. Scurt e asa. Un middleware e o functie cu `req, res si next` ca argumente, noi o putem scrie in felul dat `const myMiddleware = () => (req, res, next) => {}` si atunci noi o sa o utilizam in felul dat `myMiddleware()`, dar in cazurile normale pur si simplu: `myMiddleware`